### PR TITLE
[ fix #2208 ] Add interactive commands to help message

### LINKED ISSUE #2208: Interactive idris commands are not documented/not properly parsed
# Steps to Reproduce
Launch idris2 repl and run `:help`
# Expected Behavior
Show documentation on interactive commands
# Observed Behavior
No information about interactive commands
image

# Additional note
Idris won't tell anything about the interactive command until you get the semantics right
image

### PR DESCRIPTION
**Pull request by teggotic**
fixes [2208](https://github.com/idris-lang/Idris2/issues/2208)

### CODE REVIEW

File: src/Idris/Parser.idr [hunk 1]
```diff
@@ -1869,57 +1869,14 @@ replCmd (c :: cs)
...
-         line <- intLit
-         n <- name
-         pure (MakeCase upd (fromInteger line) n)
-  <|> do replCmd ["mw", "makewith"]
-         upd <- option False (symbol "!" $> True)
-         line <- intLit
-         n <- name
-         pure (MakeWith upd (fromInteger line) n)
-  <|> fatalError "Unrecognised command"
+addClauseCmd : Rule EditCmd
```
Comment (mattpolzin) [inline src/Idris/Parser.idr:1872]: Unless I am missing something, this function didn't end up getting used.

**Review by mattpolzin — COMMENTED** [review]
Sorry this sat for a while!

This is mostly making sense to me; are all of the commands you refactored covered by the test suite already? If not, this would be a good time to add them.

File: src/Idris/Parser.idr [hunk 2]
```diff
@@ -1959,25 +1916,45 @@ data CmdArg : Type where
      ||| The command takes a on or off.
      OnOffArg : CmdArg
 
+     ||| The command takes an argument documenting it's name
```
Comment (gallais) [inline src/Idris/Parser.idr:1919]:
```suggestion
     ||| The command takes an argument documenting its name
```

File: src/Idris/Parser.idr [hunk 3]
```diff
@@ -1959,25 +1916,45 @@ data CmdArg : Type where
...
+     NamedCmdArg : String -> CmdArg -> CmdArg
+
+     ||| The command takes an argument documenting it's default value
```
Comment (gallais) [inline src/Idris/Parser.idr:1922]:
```suggestion
     ||| The command takes an argument documenting its default value
```

**Review by gallais — COMMENTED** [review]
Exciting!

**Pull request merged**